ledger/simulation/tracer.go

```diff
@@ -163,9 +173,71 @@ func (tracer *evalTracer) saveApplyData(applyData transactions.ApplyData) {
 	applyDataOfCurrentTxn.EvalDelta = evalDelta
 }
 
+func (tracer *evalTracer) BeforeTxn(ep *logic.EvalParams, groupIndex int) {
+	if tracer.result.ExecTraceConfig > NoExecTrace {
```

Comment:
I don't like seeing less than, greater than comparisons on enum constants. I think it's best to pretend they are not even integers.  Is the idea that you have a few constants in mind, where all the higher ones do the same thing here?

Reply:
yea, I was not familiar with the convention here, though I read some code in codebase that use inequality. I will adapt to inequality rather than greater/less than.

Reply:
this enum comparison just get removed in [remove enum, flatten result and request in simulation](https://github.com/algorand/go-algorand/pull/5360/commits/59e8ff0dc826e5ff78cd242e39d84c6f4bcf8e46)